Package.swift

```diff
@@ -1,29 +1,53 @@
-// swift-tools-version:4.2
+// swift-tools-version:5.1
 
 import PackageDescription
+import Foundation
+
+#if os(Linux)
+import Glibc
+#else
+import Darwin.C
+#endif
+
+// Enforcing exclusivity increases compile time of release builds by 2 minutes.
+// Disable it.
+var swiftSyntaxUnsafeFlags: [String] = [
+  "-enforce-exclusivity=unchecked"
+]
+
+// If building SwiftSyntax as a dylib we are in a controlled CI environment.
+// Thus we can use internal compiler flags like the group info.
+if getenv("SWIFT_SYNTAX_BUILD_DYLIB") != nil {
+  let groupFile = URL(fileURLWithPath: #file)
+    .deletingLastPathComponent()
+    .appendingPathComponent("utils")
+    .appendingPathComponent("group.json")
+  swiftSyntaxUnsafeFlags += ["-Xfrontend", "-group-info-path",
+                             "-Xfrontend", groupFile.path]
+}
 
 let package = Package(
   name: "SwiftSyntax",
   targets: [
     .target(name: "_CSwiftSyntax"),
-    .target(name: "SwiftSyntax", dependencies: ["_CSwiftSyntax"]),
+    .target(name: "SwiftSyntax", dependencies: ["_CSwiftSyntax"],
+            swiftSettings: [.unsafeFlags(swiftSyntaxUnsafeFlags)]
```

Comment:
Won't the use of `unsafeFlags` prevent this target from being usable as a dependency ([SPM proposal](https://github.com/apple/swift-evolution/blob/master/proposals/0238-package-manager-build-settings.md#unsafe-flags-all))?

cc: @aciidb0mb3r

Reply:
It definitely works as a local dependency. I haven’t tried a versioned dependency yet. But thanks for pointing this out!

Reply:
Yes. `unsafeFlags` will disqualify it.

Reply:
We should use unsafe flags if an environment variable is set (which the CI will set).

Reply:
Yeah, published packages are not allowed to use unsafe flags.